### PR TITLE
add-user error messaging

### PR DESCRIPTION
**Pull request by indirectlylit**
based on https://github.com/learningequality/kolibri/pull/366

see also:
* https://trello.com/c/cEvFiTVb/368-users-bug-no-user-facing-feedback-of-errors-while-creating-editing-users-e-g-duplicate-user-error
* https://trello.com/c/1YRkWjHQ/387-make-view-model-data-access-explicit-rather-than-implicit

### CODE REVIEW

**Comment by indirectlylit** [comment]
Current behavior shows contradictory errors when trying to add a duplicate user:

image

The server is currently failing with this message:

```
ERROR    ValidationError while validating model before checking permissions: ({'__all__': [ValidationError([u'Facility user with this Username and Facility already exists.'])]}, None, None)
[11/Aug/2016 18:32:04] "POST /api/facilityuser/ HTTP/1.1" 403 63
```

We should update two things:

1. The server should respond with appropriate error codes, e.g. `403` for permission errors and `409` for this kind of conflict.
2. The client should check the error code:
  * If it's `403`, display "Your account does not have permission to do this"
  * If it's `409`, display "An account with that username already exists"
  * otherwise, fall back to generic error handling (setting the global error variable)

(cc @jamalex )

**Comment by indirectlylit** [comment]
merged updates - cc @ralphiee22

**Comment by indirectlylit** [comment]
@DXCanas Sorry this didn't get into master sooner - now it has some conflicts :-/

Think you and @ralphiee22 can get this worked out with proper messaging?

thanks!

File: kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue [hunk 1]
```diff
@@ -78,12 +78,15 @@
...
-          },
-          (error) => {
-            // Need more descriptive errors
-            this.errorMessage = `${error} Please make sure this username does not already exist.`;
-          }
-        );
+          }).catch((error) => {
+            if (error.status.code === 409) {
+              this.errorMessage = error.entity;
+            } else if (error.status.code === 403) {
```
Comment (DXCanas) [inline kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue:84]: Hmm. A little soggy @ralphiee22. ;P

**Pull request closed without merging**